descwl_coadd/coadd.py

```diff
@@ -1355,3 +1355,140 @@ def get_pbar(exps, nmin=5):
         return PBar(exps)
     else:
         return exps
+
+
+def get_coadd_psf_at_position(
+    exps,
+    coadd_wcs,
+    coadd_bbox,
+    psf_dims,
+    xy=None,
+    *,
+    skypos=None,
+    rng=None,
+    remove_poisson=False,
+    psfs=None,
+    wcss=None,
+    max_maskfrac=MAX_MASKFRAC,
+    bad_mask_planes=FLAGS2INTERP,
+    warper=None,
+    im_dtype=np.float32,
+):
+    """
+    Build the *coadd PSF* evaluated at a specified position in the coadd image.
+
+    The function reconstructs the effective PSF at a given coadd location by:
+      (1) evaluating each SE (single-epoch) PSF at the sky position corresponding
+          to the requested coadd pixel,
+      (2) warping each SE PSF image into the coadd WCS/bounding box, and
+      (3) stacking the warped PSFs with inverse-variance weights (1/medvar),
+          skipping inputs whose masked-pixel fraction exceeds ``max_maskfrac``.
+    The stacked PSF image is then normalized and returned as a ``KernelPsf``.
+
+    Parameters
+    ----------
+    exps : sequence of `lsst.afw.image.ExposureF` or `ExposureD`
+        Single-epoch exposures contributing to the coadd.
+    coadd_wcs : `lsst.afw.geom.SkyWcs`
+        WCS of the target coadd.
+    coadd_bbox : `lsst.geom.Box2I`
+        Bounding box (in coadd pixels) of the coadd cell/region.
+    psf_dims : tuple[int, int]
+        Desired PSF stamp dimensions (must be square and odd).
+    xy : tuple[float, float] or `lsst.geom.Point2D`, optional
+        Coadd pixel coordinates (x, y) at which to evaluate the PSF. If not
+        provided, you must pass ``skypos``.
+    skypos : `lsst.geom.SpherePoint`, optional
+        Sky position at which to evaluate the PSF. If provided, ``xy`` is ignored.
```

Comment:
I think these arguements should match the galsim conventions of using `image_pos` and `world_pos`, both should be optional (so before the `*` in the function signature), and the code should raise if both are passed or if neither are passed.